tablo/interfaces/arcgis/views.py

```diff
@@ -261,61 +270,67 @@ def handle_request(self, request, **kwargs):
 
         try:
             # Query with limit plus one to determine if the limit excluded any features
-            query_response = self.feature_service_layer.perform_query(int(limit) + 1, offset, **search_params)
-        except (DatabaseError, ValueError):
+            query_response = self.feature_service_layer.perform_query(limit, offset, **search_params)
+        except ValidationError as ex:
+            return HttpResponseBadRequest(json.dumps({'error': ex.message}))
+        except DatabaseError:
+            # Any generic database error, or failed validation of SQL injection
```

Comment:
Good catch. I'll adapt it and move it to the `ValidationError` block since that's where a range of different errors may be handled.